dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/merge/category/CategoryOptionMergeProcessorTest.java

```diff
@@ -239,6 +245,29 @@ void catOptComboRefsReplacedSourcesNotDeletedTest() throws ConflictException {
     assertTrue(allCategoryOptions.containsAll(List.of(coTarget3A, coSource1A, coSource2B)));
   }
 
+  @Test
+  @DisplayName("Expect the correct number of SQL delete queries when merging")
+  void catOptMergeQueryTest() throws ConflictException {
+    // given
+    CategoryCombo cc1 = createCategoryCombo('1', cat1);
+    CategoryCombo cc2 = createCategoryCombo('2', cat3);
+    manager.save(List.of(cc1, cc2));
+
+    categoryService.updateOptionCombos(cc1);
+    categoryService.updateOptionCombos(cc2);
+
+    // when
+    MergeParams mergeParams = getMergeParams();
+    mergeParams.setSources(Set.of(UID.of(coSource1A)));
+    mergeParams.setDeleteSources(true);
+
+    SQLStatementCountValidator.reset();
+    mergeProcessor.processMerge(mergeParams);
+
+    // then
+    assertDeleteCount(1);
```

Comment:
- Could we add another check here, such as `mergeReport.sourcesDeleted == 1` ? In case the library you just added doesn't work correctly in the future maybe cause of referenced libraries get upgraded.

Reply:
sure, i'll add a call to get the object as well. The inclusion in the report wouldn't guarantee that it's deleted either.